source/FreeRTOS_Stream_Buffer.c

```diff
@@ -174,24 +159,18 @@ void vStreamBufferClear( StreamBuffer_t * pxBuffer )
  * @param[in] pxBuffer The circular stream buffer.
  * @param[in] uxCount The byte count by which the mid pointer is to be moved.
  */
-void vStreamBufferMoveMid( StreamBuffer_t * pxBuffer,
-                           size_t uxCount )
+void vStreamBufferMoveMid( StreamBuffer_t * const pxBuffer,
+                           const size_t uxCount )
 {
     /* Increment uxMid, but no further than uxHead */
-    size_t uxSize = uxStreamBufferMidSpace( pxBuffer );
-    size_t uxMid = pxBuffer->uxMid;
-    size_t uxMoveCount = uxCount;
+    const size_t uxSize = uxStreamBufferMidSpace( pxBuffer );
+    const size_t uxMoveCount = ( uxCount > uxSize ) ? uxSize : uxCount;
```

Comment:
I prefer the original implementation, which was easier to read. Conditional operators make the code "more dense", and harder to debug.